src/components/admin/ContentRenderer.tsx

```diff
@@ -77,11 +79,13 @@ export const ContentRenderer = async ({
     markAsCompleted: boolean;
     bookmark: Bookmark | null;
   };
+  courseId: number;
 }) => {
   const metadata = await getMetadata(content.id);
-
+  const prevNextUrls = await getNextPrevUrls(content.id, courseId);
```

Comment:
dont do this here asynchronously. Thisll delay rendering until this is fetched, and this isnt needed for most renders

Do it in an effect maybe?

Reply:
sir using cached values instead of database fetches.
please do check the last file `./db/course.ts`